CHANGELOG.md

```diff
@@ -1,7 +1,75 @@
 Change Log
 ==========
 
-# Version v4.0.0
+# Version 4.0.1
+
+_2024-10-01_
+
+This release contains a handful of bug fixes and improvements. 
+
+## 🧩 Compiler Plugins additions (experimental)
+
+The [compiler plugins](https://www.apollographql.com/docs/kotlin/advanced/compiler-plugins) API has been extended to allow listening to schema changes.\
+This can be used to have plugins generate code based on the schema. To do this, implement the [`schemaListener`](https://www.apollographql.com/docs/kotlin/kdoc/apollo-compiler/com.apollographql.apollo.compiler/-apollo-compiler-plugin/schema-listener.html)
+function on your plugin:
+    
+```kotlin
+class MyCompilerPlugin() : ApolloCompilerPlugin {
+  @ApolloExperimental
+  override fun schemaListener(): SchemaListener {
+    return object : SchemaListener {
+      override fun onSchema(schema: Schema, outputDirectory: File) {
+        // Generate some code inside outputDirectory based on schema
+      }
+    }
+  }
+}
+```
+
+## ♻️ Deprecations
+- The `downloadApolloSchema` task is deprecated, please use an [`introspection` block](https://go.apollo.dev/ak-download-schema) to download a schema.
+- The `generateOperationOutput` property is deprecated, please use [`operationManifestFormat.set("persistedQueryManifest")`](https://www.apollographql.com/docs/kotlin/advanced/persisted-queries/#1-generate-operation-manifest) instead".
+- The `@nonnull` directive is deprecated, please use [`@semanticNonNull` / `@catch`](https://go.apollo.dev/ak-nullability) instead.
```

Comment:
Do we usually mention deprecations? I know some project do it (Gradle maybe?) but it's small signal to me as I get the information through other means (compiler warnings mostly).

Reply:
Yeah we haven't done it before actually - ok to remove the section.

Reply:
Mind adding this rule to the CONTRIBUTING.md for reference?